src/ImageStats/Histogram.h

```diff
@@ -18,25 +18,33 @@ struct HistogramResults {
     int num_bins;
     float bin_width;
     float bin_center;
-    std::vector<int> histogram_bins;
+    std::vector<int> histogram_bins; // histogram bin counts
 };
 
 class Histogram {
-    int _num_bins;
-    float _bin_width;
-    float _min_val;
-    std::vector<int> _hist;
+    float _min_val; // lower bound of the histogram (inclusive)
+    float _max_val; // upper bound of the histogram (inclusive)
+    HistogramResults _histogram; // histogram stats
     const std::vector<float>& _data;
 
+    bool ConsistencyCheck(const Histogram&, const Histogram&);
+
 public:
     Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data);
     Histogram(Histogram& h, tbb::split);
 
     void operator()(const tbb::blocked_range<size_t>& r);
-    void join(Histogram& h); // NOLINT
+    bool join(Histogram& h); // NOLINT
     void setup_bins();
 
-    HistogramResults GetHistogram() const;
+    HistogramResults GetHistogram() const { return _histogram; }
+    float GetMinVal() const { return _min_val; }
+    float GetMaxVal() const { return _max_val; }
+    int GetNbins() const { return _histogram.num_bins; }
+    float GetBinWidth() const { return _histogram.bin_width; }
+    float GetBinCenter()  const { return _histogram.bin_center; }
+    const std::vector<int>& GetHistogramBins() const { return _histogram.histogram_bins; }
+    const std::vector<float>& GetData() const { return _data; }
```

Comment:
I don't think we need to hang on to the data after we've constructed the histogram

Reply:
@veggiesaurus, I'd actually drop the data from the class. After calling setup_bins in the constructor, we don't need it anymore.

Reply:
Exactly